articles/arm-x86/index.md

```diff
@@ -0,0 +1,85 @@
+
+## Drawbacks of ARM
+
+There is an issue of compatibility here though. Programs will need to be completely rewritten to run on Apple's newer machines. This isn't too bad, since most compilers can target ARM just fine. Assembly developers are probably having a panic attack right now. Apple is also hoping that [Rosetta 2](https://www.theverge.com/21304182/apple-arm-mac-rosetta-2-emulation-app-converter-explainer) can run all the x86 applications. Rosetta 1 was known for being much slower than running on original hardware. This is by necessity. Rosetta needs to translate the x86 instructions into ARM instructions in real-time. To be fair, Java converts from [bytecode to everything else](https://www.javatpoint.com/java-bytecode), and it seems to work [fine-ish](https://benchmarksgame-team.pages.debian.net/benchmarksgame/fastest/java.html). Still, it seems hard to imagine being able to play games from the 2010's if Microsoft also decides to switch to ARM.
+
+There's also an issue of speed. Since ARM has fewer instructions, developers need to use more instructions. For example, did you know that [ARM usually doesn't have a division instruction](https://cseweb.ucsd.edu/classes/wi14/cse30-c/lectures/PI_WI_14_CSE30_lecture_8_post.pdf)? You have to divide by using [other instructions](https://stackoverflow.com/questions/19844575/how-to-do-division-in-arm). Since you're using other instructions to fake division, it ends up taking many cycles. This could even be slower than on a CISC instruction set.
+
+### Executable Sizes
+
+Since you need more instructions in ARM, it's possible that executables will be larger. To test this, I compiled the [sorting_algos repository](https://github.com/botahamec/sorting_algos) made for our [sorting article.](https://www.section.io/engineering-education/sorting-algorithms/) I ran it on my [Raspberry Pi 4 Model B Rev 1.1](https://www.raspberrypi.org/products/raspberry-pi-4-model-b/). I compiled the same thing on my x86-64 Linux desktop. The Raspberry Pi file size should actually be a little smaller. It's only using a 32-bit operating system (Raspberry Pi OS Lite). So, I cross-compiled to the stable-i686-unknown-linux-gnu toolchain on my desktop. "Stripped" means that I stripped nonessential symbols from the executable. The Cargo profiles I used are at the bottom. Here are the results:
```

Comment:
You can put this under a "Tutorial" subheading and of course remove all mentions of "I" or first-person writing. Rather, explain the exercise and what was accomplished after testing. Describe the steps as in a tutorial so that if a reader wanted to follow along and test it for themselves, they could.

Reply:
I removed the "I".

Like division algorithms, I think explaining how to target ARM in Rust is very far from the purpose of the article. I could include a link to a list of steps, but there's no point in making all of the readers sit through a tutorial that very few of them want to follow along with.